dotcom-rendering/src/model/enhance-H3s.ts

```diff
@@ -0,0 +1,92 @@
+import { JSDOM } from 'jsdom';
+
+const isFalseH3 = (element: CAPIElement): boolean => {
+	if (!element) return false;
+	// Checks if this element is a 'false h3' based on the convention: <p><strong><H3 text</strong></p>
+	if (
+		element._type !== 'model.dotcomrendering.pageElements.TextBlockElement'
+	) {
+		return false;
+	}
+	const frag = JSDOM.fragment(element.html);
+	if (!frag || !frag.firstElementChild) return false;
+	const html = frag.firstElementChild.outerHTML;
+	// The following things must be true for an element to be a faux H3
+	const hasPwrapper = frag.firstElementChild.nodeName === 'P';
+	const containsStrongtags =
+		frag.firstElementChild.outerHTML.includes('<strong>');
+	const doesNotContainLinks =
+		!frag.firstElementChild.outerHTML.includes('<a>');
+	const startStrong = html.startsWith('<p><strong>');
+	const endsStrong = html.endsWith('</strong></p>');
+
+	return (
+		hasPwrapper &&
+		containsStrongtags &&
+		doesNotContainLinks &&
+		startStrong &&
+		endsStrong
+	);
```

Comment:
If you want to avoid unncessary work, you can follow the pattern of early returns:

```ts
const frag = JSDOM.fragment(element.html);
if (!frag.firstElementChild) return false;

// The following things must be true for an element to be a faux H3
if (frag.firstElementChild.nodeName !== 'P') return false;
if (!frag.firstElementChild.outerHTML.includes('<strong>')) return false;
if (frag.firstElementChild.outerHTML.includes('<a>')) return false;

// note: we don’t need to extract html until now.
const html = frag.firstElementChild.outerHTML;
if (!html.startsWith('<p><strong>')) return false;
if (!html.endsWith('</strong></p>')) return false;

return true;
```

Reply:
I think the first version is easier to read. 

        hasPwrapper &&
		containsStrongtags &&
		doesNotContainLinks &&
		startStrong &&
		endsStrong

Makes it very clear what needs to be true for the element to be considered an h3.

I believe a list of ORed negations is much more difficult to get one's head around, hence the need for the comment there. 

Since the performance gain in this case it negligible I would prefer keeping the original version if you don't mind?

In general I do agree returning early is preferable.

Reply:
Agree, readability trumps micro-performance optimisations.

Keep it as-is.